crates/wasmtime/build.rs

```diff
@@ -92,3 +97,26 @@ fn build_c_helpers() {
     build.file("src/runtime/vm/helpers.c");
     build.compile("wasmtime-helpers");
 }
+
+fn enable_features_based_on_rustc_version() {
+    // Temporary check to see if the rustc version >= 1.84 in which case
+    // provenance-related pointer APIs are available. This is temporary because
+    // in the future the MSRV of this crate will be beyond 1.84 in which case
+    // this build script can be deleted.
+    let minor = rustc_minor_version().unwrap_or(0);
+    if minor >= 84 {
+        println!("cargo:rustc-cfg=has_provenance_apis");
+    }
+    println!("cargo:rustc-check-cfg=cfg(has_provenance_apis)");
+}
+
+fn rustc_minor_version() -> Option<u32> {
+    let rustc = std::env::var("RUSTC").unwrap();
+    let output = Command::new(rustc).arg("--version").output().ok()?;
+    let version = str::from_utf8(&output.stdout).ok()?;
+    let mut pieces = version.split('.');
```

Comment:
Nit: Could use `version.split_once('.')?` here (and in pulley/build.rs) to simplify the code a bit.

Reply:
Sort of but it'd end up requiring two invocations: one for looking at `rustc 1` and the next for looking at the N in `rustc 1.N.0`. Given that I think I'll leave it as `.split('.')`, but I'm also just [copy/pasting this detection code around](https://github.com/bytecodealliance/wasm-tools/blob/7530629a3a127a1429b4837a1b1632f11d3dfeea/crates/wasmparser/build.rs#L18-L27) so I could also be missing something